src/bucketd/processor.rs

```diff
@@ -282,30 +282,59 @@ impl Runtime {
 
                         let mut owned_rights: BTreeMap<OwnedRightType, TypedAssignments> = bmap! {};
                         for (owned_type, assignments) in transition.owned_rights().iter() {
-                            let outpoints = assignments.to_value_assignments();
-
-                            let mut revealed_assignment: Vec<Assignment<PedersenStrategy>> =
-                                empty!();
-
-                            for out in outpoints {
-                                if out.commit_conceal().to_confidential_seal()
-                                    != reveal_outpoint.to_concealed_seal()
-                                {
-                                    revealed_assignment.push(out);
-                                } else {
-                                    let accept = match out.as_revealed_state() {
-                                        Some(seal) => Assignment::Revealed {
-                                            seal: reveal_outpoint,
-                                            state: *seal,
-                                        },
-                                        _ => out,
-                                    };
-                                    revealed_assignment.push(accept);
+                            match assignments {
+                                TypedAssignments::Value(outpoints) => {
+                                    let mut assignment: Vec<Assignment<PedersenStrategy>> =
+                                        empty!();
+
+                                    for out in outpoints.clone() {
```

Comment:
It will be more efficient to remove `clone` here and put in in just a single branch which takes the ownership below (the other branch doesn't need to do the clone).